src/rust-2024/rustdoc-doctests.md

```diff
@@ -0,0 +1,76 @@
+# Rustdoc combined tests
+
+🚧 The 2024 Edition has not yet been released and hence this section is still "under construction".
+More information may be found in the tracking issue at <https://github.com/rust-lang/rust/issues/124853>.
+
+## Summary
+
+- [Doctests] are now combined into a single binary which should result in a significant performance improvement.
+
+## Details
+
+Prior the the 2024 Edition, rustdoc's "test" mode would compile each code block in your documentation as a separate executable. Although this was relatively simple to implement, it resulted in a significant performance burden when there are a large number of documentation tests. Starting with the 2024 Edition, rustdoc will attempt to combine documentation tests into a single binary, significantly reducing the overhead for compiling doctests.
+
+```rust
+/// Adds two numbers
+///
+/// ```
+/// assert_eq!(add(1, 1), 2);
+/// ```
+pub fn add(left: u64, right: u64) -> u64 {
+    left + right
+}
+
+/// Subtracts two numbers
+///
+/// ```
+/// assert_eq!(subtract(2, 1), 1);
+/// ```
+pub fn subtract(left: u64, right: u64) -> u64 {
+    left - right
+}
+```
+
+In this example, the two doctests will now be compiled in a single executable. Rustdoc will essentially place each example in a separate function within a single binary. Rustdoc also adds a small `main` function which tells it which test to run. The tests still run in independent processes as they did before, so any global state (like global statics) should still continue to work correctly.
```

Comment:
I'm not sure that explaining that rustdoc adds a `main` function is really useful here. Especially since users can add their own `main` function in the doctest directly. I think the sentence can be removed. Just in case, talking about this one:

```
Rustdoc also adds a small `main` function which tells it which test to run.
```

Reply:
Yea, I was a bit on the fence on how to describe how it works. I was feeling it would be useful to give the reader a better understanding of how things work under the hood. But overall it probably isn't all that important so I have removed it.

Reply:
Hum... If it's something people are interested into, would it be ok to link to an external blog post? I explained more in details how it works [here](https://blog.guillaume-gomez.fr/articles/2024-08-17+Doctests+-+How+were+they+improved%3F).

Reply:
Sure!